src/lib/functools.js

```diff
@@ -0,0 +1,459 @@
+function $builtinmodule() {
+    const functools = {
+        __name__: new Sk.builtin.str("functools"),
+        __doc__: new Sk.builtin.str("Tools for working with functions and callable objects"),
+        __all__: new Sk.builtin.list(
+            [
+                "update_wrapper",
+                "wraps",
+                "WRAPPER_ASSIGNMENTS",
+                "WRAPPER_UPDATES",
+                "total_ordering",
+                "cmp_to_key",
+                "lru_cache" /**@todo lru_cache */,
+                "reduce",
+                "TopologicalSorter" /**@todo TopologicalSorter */,
+                "CycleError" /**@todo CycleError */,
+                "partial",
+                "partialmethod",
+                "singledispatch" /**@todo singledispatch */,
+                "singledispatchmethod" /**@todo singledispatchmethod */,
+                "cached_property" /**@todo cached_property */,
+            ].map((x) => new Sk.builtin.str(x))
+        ),
+        WRAPPER_ASSIGNMENTS: new Sk.builtin.tuple(
+            ["__module__", "__name__", "__qualname__", "__doc__" /*"__annotations__"*/].map((x) => new Sk.builtin.str(x))
+        ),
+        WRAPPER_UPDATES: new Sk.builtin.tuple([new Sk.builtin.str("__dict__")]),
+
+        /**@todo */
+        lru_cache: proxyFail("lru_cace"),
```

Comment:
typo?? should be `lru_cache` ??

PS - would be very cool to have this functionality!

Reply:
yes it would! I've never used those tools personally so it wasn't something I needed - when this gets merged I'll report an issue for the missing features for someone (or me) to have a go at.  will fix the typo